Translate yggdrasil.md (#17)
LittleSkinCommspt/LittleSkinManualDev: en/advanced/yggdrasil.md

# Yggdrasil

> LittleSkin provides Yggdrasil Verify Authentication Service. You can authenticate users when they join servers and load skins from LittleSkin without skin loader mods by combining using with [authlib-injector](https://github.com/yushijinhun/authlib-injector).

Reviewer: ```suggestion
LittleSkin provides Yggdrasil Authentication Service. You can authenticate users when they join servers and load skins from LittleSkin without skin loader mods by using the [authlib-injector](https://github.com/yushijinhun/authlib-injector).
```

Reviewer 2: >authlib-injector 不是特指，不需要加 the
根据邱的建议，我认为不应该对 authlib 加 the

You can download the latest version of authlib-injector at <https://authlib-injector.yushi.moe/~download/>

## Yggdrasil API root

The Yggdrasil API root of LittleSkin is:

```
https://littleskin.cn/api/yggdrasil
```

> LittleSkin supports API Location Indication (ALI) feature of authlib-injector, so it's okay to just enter any URL of LittleSkin when using launchers and authlib-injector supports ALI and it will be identified.

Reviewer: ```suggestion
LittleSkin supports API Location Indication (ALI) feature of authlib-injector, so it's okay to just enter any URL of LittleSkin when using launchers and authlib-injector with ALI supports and the URL could be identified.
```

## API Documents

See: [Yggdrasil Server Technical Specifications](https://github.com/yushijinhun/authlib-injector/wiki/Yggdrasil%E6%9C%8D%E5%8A%A1%E7%AB%AF%E6%8A%80%E6%9C%AF%E8%A7%84%E8%8C%83)

## Using in the server

Please set `online-mode` to `true` in your server configuration file (typically `server.properties`) first. Then add the following parameter to your startup command before the `-jar` parameter:

```
-javaagent:{path/to/authlib-injector.jar}=https://littleskin.cn/api/yggdrasil
```

> The `{path/to/authlib-injector.jar}` is the path that directs to the authlib-injector.

Reviewer: ```suggestion
The `{path/to/authlib-injector.jar}` is the directory of the authlib-injector.
```

::: warning

> <strong>DO NOT include parentheses at either end!<strong> The big parentheses are only used to show that you must assign this value correctly, but needn't add the big paratheses in real.

Reviewer: ```suggestion
<strong>DO NOT include parentheses at either end!<strong> The curly parentheses are only used to show that you must assign this value correctly but you needn't add the curly paratheses in real.
```

怎么想这也不是能直译的

```
-javaagent:{path/to/authlib-injector.jar}=https://littleskin.cn/api/yggdrasil
```

其中 `{path/to/authlib-injector.jar}` 为指向 authlib-injector 的路径。
In this, `{path/to/authlib-injector.jar}` is the path that directs to authlib-injector.

::: warning
<strong>实际填写的路径两边没有大括号！</strong>大括号只是表明必须正确指定这个参数的值，但是实际上并不需要填入大括号。
<strong>The path you write doesn’t have big parentheses!<strong>The big parentheses are only shown that must be assigned this value correctly, but needn't write big paratheses in real.
:::

::: tip
如果你使用 BungeeCord，你需要在所有服务端（包括 BungeeCord 和所有子服）中加载 authlib-injector（方法见上），但只有 BungeeCord 可以打开 `online-mode`，其他服务端应该关闭 `online-mode`。
If you're using BungeeCord, you need to load authlib-injector for all servers (including BungeeCord and all sub servers), but you can only open `online-mode` of BungeeCord, keep closed for other servers.
:::

> ## 在客户端使用

Reviewer 2: Us

> You need a launcher which supports using a custom Yggdrasil server so that you can use the Yggdrasil service of LittleSkin in the client. [HMCL 3](https://www.mcbbs.net/thread-142335-1-1.html) and [BakaXL](https://www.mcbbs.net/thread-512144-1-1.html) is recommended.

Reviewer: ```suggestion
You need a launcher which supports using a custom Yggdrasil server so that you can use the Yggdrasil service of LittleSkin in the client. [HMCL 3](https://www.mcbbs.net/thread-142335-1-1.html) and [BakaXL](https://www.mcbbs.net/thread-512144-1-1.html) are recommended.
```

> Here we use HMCL 3 to demonstrate how to use LittleSkin's Yggdrasil API in a Minecraft client:

Reviewer: ```suggestion
Here's a demo about how to use LittleSkin's Yggdrasil API in a Minecraft client based on HMCL3:
```

![set-login-method](./assets/yggdrasil/set-login-method.png)

2. Click the "+" button at the right of "Authentication Server", enter the Yggdrasil API root of LittleSkin at the text field, and click "Next". Please wait a moment for the launcher to retrieve information from our website. Once finished, click "Finish".
If you have added our Yggdrasil API before, simply select LittleSkin from the authentication server list.

![add-yggdrasil-server-1](./assets/yggdrasil/add-yggdrasil-server-1.png)

![add-yggdrasil-server-2](./assets/yggdrasil/add-yggdrasil-server-2.png)

![add-yggdrasil-server-3](./assets/yggdrasil/add-yggdrasil-server-3.png)

3. Select LittleSkin from the authentication server list and type your email address and password. "Name" is your email address bound to your LittleSkin account.

> If you only add an auth server, HMCL 3 will select the only auth server by default.

Reviewer: ```suggestion
If you have only one auth server, HMCL 3 will select that auth server by default.
```

![set-email-password](./assets/yggdrasil/set-email-password.png)

4. Select your player's profile.

> When there's only one player's profile in the account, that profile will be the default choice.

Reviewer: ```suggestion
When there's only one player's profile in the account, that profile will be chosen by default.
```

If you wanna add more profiles, you can add your account again to add another profile.

![choose-player](./assets/yggdrasil/choose-player.png)

5. Setup is complete! You can start the game now.

::: tip A point
For PCL2, you might need to configure the Yggdrasil server address by yourself at `Version settings` → `Settings` → `Server options`.

Login type：`Authlib-Injector`
Auth server：`https://littleskin.cn/api/yggdrasil`
Register address： `https://littleskin.cn/auth/register`
Server name：`LittleSkin`
:::

> ## Reference links

Reviewer 3: ```suggestion
## Reference
```

上次讲过，Reference 作名词时本身就有“参考内容的出处”的意思，没有必要再加 link
